aws-s3-transfer-manager/src/operation/upload/handle.rs

```diff
@@ -12,15 +12,20 @@ use aws_sdk_s3::error::DisplayErrorContext;
 use aws_sdk_s3::types::{CompletedMultipartUpload, CompletedPart};
 use tokio::sync::Mutex;
 use tokio::task;
+use tracing::Instrument;
 
 /// Response type for a single upload object request.
 #[derive(Debug)]
 #[non_exhaustive]
 pub struct UploadHandle {
     /// All child multipart upload tasks spawned for this upload
     pub(crate) upload_tasks: Arc<Mutex<task::JoinSet<Result<CompletedPart, crate::error::Error>>>>,
+    /// Parent tracing span for spawned upload tasks.
+    pub(crate) parent_span_for_upload_tasks: tracing::Span,
     /// All child read body tasks spawned for this upload
     pub(crate) read_tasks: task::JoinSet<Result<(), crate::error::Error>>,
+    /// Parent tracing span for spawned read body tasks.
+    pub(crate) parent_span_for_read_tasks: tracing::Span,
```

Comment:
Do we need to store these within the handle? Can we just create them within the distribute_work function and use them there?

Reply:
Similar for download, instead of storing it in the handle, can we just use it in the distribute_work function?

Reply:
oh hey, you're right